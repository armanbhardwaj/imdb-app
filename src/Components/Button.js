import React from 'react';
// import '../../src/';

const Button = (props) =>
{
    return( <button className={props.className} onClick={props.onClick}>{props.children}</button>);
}
export default Button;