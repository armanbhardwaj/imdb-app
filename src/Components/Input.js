import React from 'react';

const Input =(props)=>
{
    return( <input placeholder={props.placeholder} className={props.className} name={props.name} onChange={props.onChange} type={props.type} value={props.value} />);
       
}
export default Input;