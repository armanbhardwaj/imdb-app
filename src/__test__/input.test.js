import Input from '../Components/Input';
import Button from '../Components/Button';
import { shallow } from 'enzyme';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('Input',()=> {
    it('checking...', ()=>{
        expect(Input).toBeDefined();
    })
    it('input should render', ()=>{
        const tree = shallow(<Input name="arman" value="bhardwaj" placeholder="enter here" />)
        expect(tree).toMatchSnapshot();
    })
})

describe('Button', ()=>{
    it('checking...', ()=>{
        expect(Input).toBeDefined();
    })
    it('button should render', ()=>{
        const tree =shallow(<Button name="vivian" value="dsena"/>)
        expect(tree).toMatchSnapshot();
    })
})