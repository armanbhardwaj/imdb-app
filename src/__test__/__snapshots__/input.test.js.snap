// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="vivian"
    value="dsena"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Input input should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    name="arman"
    placeholder="enter here"
    value="bhardwaj"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": undefined,
      "name": "arman",
      "onChange": undefined,
      "placeholder": "enter here",
      "type": undefined,
      "value": "bhardwaj",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": undefined,
        "name": "arman",
        "onChange": undefined,
        "placeholder": "enter here",
        "type": undefined,
        "value": "bhardwaj",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
